runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -211,6 +211,361 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+ * Each element is 1-byte for compact strings and 2-bytes for non-compressed strings.
+ *
+ * \details
+ *
+ * The vector sequence searches for the first character of the sub-string inside the source/master string. If the first character
+ * is located, it'll perform iterative vector binary compares to match the rest of the sub-string starting from the first character position.
+ *
+ * This evaluator inlines the following Java intrinsic methods:
+ *
+ * <verbatim>
+ * For Java 9 and above:
+ *
+ * StringLatin1.indexOf(s1Value, s1Length, s2Value, s2Length, fromIndex);
+ * StringUTF16.indexOf(s1Value, s1Length, s2Value, s2Length, fromIndex);
+ *
+ * For Java 8:
+ * com.ibm.jit.JITHelpers.intrinsicIndexOfStringLatin1(char[] s1Value, int s1len, char[] s2Value, int s2len, int start);
+ * com.ibm.jit.JITHelpers.intrinsicIndexOfStringUTF16(char[] s1Value, int s1len, char[] s2Value, int s2len, int start);
+ *
+ * Assumptions:
+ *
+ * -# 0 <= fromIndex < s1Length
+ * -# s1Length could be anything: positive, negative or 0.
+ * -# s2Length > 0
+ * -# s1Value and s2Value are non-null arrays and are interpreted as byte arrays.
+ * -# s1Length and s2Length are not related. i.e. s1Length could be smallers than s2Length.
+ *
+ * <\verbatim>
+ *
+ * \param node the intrinsic function call node
+ * \param cg the code generator
+ * \param isCompressed true if the string is compressed.
+ *
+ * \return a register for that contains the indexOf() result.
+*/
+TR::Register*
+inlineVectorizedStringIndexOf(TR::Node* node, TR::CodeGenerator* cg, bool isCompressed)
+   {
+   // element size is 2-byte if both s1 and s2 are decompressed strings and it is 12 otherwise.
+   // 0 for byte, 1 for halfword
+   const uint32_t elementSizeMask = isCompressed ? 0 : 1;
+   const bool isUTF16 = !isCompressed;
+
+   const int8_t vectorSize = cg->machine()->getVRFSize();
+   const uintptrj_t headerSize = TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
+   TR::Compilation* comp = cg->comp();
+
+   if (comp->getOption(TR_TraceCG))
+      traceMsg(comp, "inlineVectorizedStringIndexOf. is compressed %d\n", isCompressed);
+
+   // Get call parameters
+   // s1Value and s2Value are byte arrays
+   // This evaluator function handles different indexOf() instrinsics, some of which are static calls without a receiver. Hence, the need for static call check.
+   const bool isStaticCall = node->getSymbolReference()->getSymbol()->castToMethodSymbol()->isStatic();
+   const uint8_t firstCallArgIdx = isStaticCall ? 0 : 1;
+
+   TR::Register* r_s1Value   = cg->evaluate          (node->getChild(firstCallArgIdx));
+   TR::Register* r_s1Len     = cg->gprClobberEvaluate(node->getChild(firstCallArgIdx+1));
+   TR::Register* r_s2Value   = cg->evaluate          (node->getChild(firstCallArgIdx+2));
+   TR::Register* r_s2Len     = cg->gprClobberEvaluate(node->getChild(firstCallArgIdx+3));
+   TR::Register* r_fromIndex = cg->gprClobberEvaluate(node->getChild(firstCallArgIdx+4));
```

Comment:
I suggest we don't deviate from the naming scheme of the rest of the codebase. In particular I don't think we've ever used Hungarian notation for local variables. Let's try not to deviate too much and perhaps use a "Reg" suffix (or "Register" which seems the be the next most common though needlessly longer):

https://github.com/eclipse/openj9/blob/8d06ec3edc1c5ad262b9d6221974a410ae20220b/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L1527-L1528

https://github.com/eclipse/openj9/blob/8d06ec3edc1c5ad262b9d6221974a410ae20220b/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L1841

https://github.com/eclipse/openj9/blob/8d06ec3edc1c5ad262b9d6221974a410ae20220b/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L1897

https://github.com/eclipse/openj9/blob/8d06ec3edc1c5ad262b9d6221974a410ae20220b/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L2228-L2229

https://github.com/eclipse/openj9/blob/8d06ec3edc1c5ad262b9d6221974a410ae20220b/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L2228-L2229

etc.

Reply:
Similar comments for labels apply.

Reply:
Changed all variables names.